### ISSUE TITLE
Add support for Babel's --out-file-extension

### ISSUE THREAD

**Issue by FezVrasta** [karlprieb/babel-plugin-add-import-extension#3]
Babel recently introduced a new flag, `--out-file-extension`, that allows to set the extension of the generated files.

This plugin doesn't seem to take it into account, so if I use `--out-file-extension=.mjs` but my source files are `.js` the imports will refer to the `.js` files instead of the `.mjs` ones.

**Comment by karlprieb** [comment]
I didn't understand how can we use that on this plugin? There's and API for that or it's only a CLI parameter?

**Comment by FezVrasta** [comment]
It's only a CLI parameter, I'm not sure if there's any way to read babel-cli arguments from a plugin?

**Comment by karlprieb** [comment]
@FezVrasta I didn't find a way to read cli arguments from plugin :(

I'll keep looking at the docs to try to add this feature!

**Comment by FezVrasta** [comment]
Would read `process.argv` bad?

**Comment by karlprieb** [comment]
I read that we can know who called `babel`, so if we know that `@babel/cli` was the one who called I think that we can use `process.argv` to read the `--out-file-extension` argument, what you think about that? It's safe?

**Comment by FezVrasta** [comment]
I'm not sure, I haven't a lot of experience with Babel :-(

**Comment by karlprieb** [comment]
I think we have the information about who called the script on the `state.opts`, I will take a look later.

**Comment by Banou26** [comment]
Would this also allow us to transform `.ts` imports to `.js` too ?
If it does then i'd gladly spend time trying to make this work

**Comment by karlprieb** [comment]
@Banou26 I don't think so, I think this would be the responsibility of this plugin, but as you may already know we skip modules that already have an extension, but you can implement that as an option, idk some boolean like `transformImportedModuleExtension` and when true change that.

**Comment by Banou26** [comment]
Would it be okay with me adding that option then ?
Or would you like to keep this plugin as minimal as it is ?

**Comment by karlprieb** [comment]
@Banou26 I think that "option" is related to import statements and file extensions and this plugin are about that, so I don't have any problem with this feature. You can send a PR with that if you want :)

Thank you for your contribution on this project so far!
